Makefile

```diff
@@ -92,7 +92,9 @@ test-integration: ## Run integration tests
 .PHONY: test-e2e
 test-e2e: ## Run e2e tests
```

Comment:
Should we add this target to `scripts/ci-test.sh`?

Reply:
How long is the runtime? If it's significant we might want to adopt it as a post-submit job rather than a pre-submit job.

Reply:
Currently, when I run `time make test-e2e` I get:
> real    2m37.973s
user    0m12.329s
sys     0m8.935s

As more e2e tests are added in, it could take longer. So post-submit 🤷‍♂ ? I really wish I knew more about the CI system because I'm not sure what the best option is.

Reply:
How stable is the test? Does it flake at all?

Reply:
I ran the test with `-count=5` on my machine and it passed without any flakes. But I know from experience that it could be different in a CI environment with constrained resources.

Currently, I've added specific check conditions to ensure that the components are up and running before the test actually runs so I'm confident that there is a low chance for flakes.

If there is concern that there will be additional burden in debugging CI test failures I'm more than happy to share that load. I remember it being brought up during a CAPI meeting some time ago.

Reply:
I think with the running time we are talking about doing it during pre-submit should be fine. We can always break it out later if needed.

Reply:
I created this issue to track this work. https://github.com/kubernetes-sigs/cluster-api/issues/1884